src/nvim/terminal.c

```diff
@@ -639,6 +639,15 @@ static void buf_set_term_title(buf_T *buf, char *title)
                &err);
 }
 
+// Get the associated buffer for the terminal and return whether it's in a
+// valid state to be used
+static bool term_buf_valid(Terminal *term, buf_T **buf)
+  FUNC_ATTR_NONNULL_ALL
+{
+  *buf = handle_get_buffer(term->buf_handle);
+  return buf_valid(*buf);
```

Comment:
perhaps return NULL in the false case, or `buf` in the true case. Then we don't need the `**buf` parameter.